orderer/consensus/etcdraft/chain.go

```diff
@@ -900,6 +900,11 @@ func (c *Chain) catchUp(snap *raftpb.Snapshot) error {
 	if c.lastBlock.Header.Number >= b.Header.Number {
 		c.logger.Warnf("Snapshot is at block [%d], local block number is %d, no sync needed", b.Header.Number, c.lastBlock.Header.Number)
 		return nil
+	} else if b.Header.Number == c.lastBlock.Header.Number+1 {
```

Comment:
I definitely don't love this, though it is fairly obviously correct, it injects knowledge of the sync properties of the WriteBlock call into the Raft implementation.